alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrRequestFactory.java

```diff
@@ -65,9 +69,35 @@ private static KeyStore loadKeyStore(String keystoreType, String file, char[] pa
             throws KeyStoreException, IOException, CertificateException, NoSuchAlgorithmException {
 
         KeyStore keyStore = KeyStore.getInstance(keystoreType);
-        try (InputStream in = new FileInputStream(ResourceUtils.getFile(file))) {
+        try (InputStream in = getResourceURL(file).openStream()) {
             keyStore.load(in, password);
         }
         return keyStore;
     }
+
+    private static URL getResourceURL(String resource) {
+        Objects.requireNonNull(resource, "Argument 'resource' is required");
+
+        // This should resolve the issue where the embedded keystore & truststore are
+        // not found on the classpath using ResourceUtils.getFile(), because the current thread has a different classloader
+
+        try {
+            if (resource.startsWith("classpath:")) {
+                String path = resource.substring("classpath:".length());
+                ClassPathResource cpResource = new ClassPathResource(path, SolrRequestFactory.class.getClassLoader());
+                if (cpResource.exists()) {
+                    return cpResource.getURL();
+                }
+            }
+        } catch (IOException e) {
+
```

Comment:
`logger.warn()` something?

Reply:
moved catch closer to `.getUrl()` and rethrowing `UncheckedIOException` now